packages/core/src/util/iitmHook.js

```diff
@@ -0,0 +1,64 @@
+/*
+ * (c) Copyright IBM Corp. 2024
+ */
+
+'use strict';
+
+const iitmHook = require('import-in-the-middle');
+
+/** @type {import('../logger').GenericLogger} */
+let logger = require('../logger').getLogger('util/iitmHook', newLogger => {
+  logger = newLogger;
+});
+
+/** @type {Object.<string, Function[]>} */
+const byModuleNameTransformers = {};
+
+/**
+ * Initializes the import-in-the-middle hooking.
+ */
+exports.init = function init() {
+  Object.entries(byModuleNameTransformers).forEach(([moduleName, applicableTransformers]) => {
+    if (applicableTransformers) {
+      applicableTransformers.forEach(transformerFn => {
+        if (typeof transformerFn === 'function') {
+          try {
+            // @ts-ignore
+            iitmHook([moduleName], (exports, name) => {
+              logger.debug(`iitm-hooking enabled for module ${name}`);
+              if (exports && exports.default) {
+                exports.default = transformerFn(exports.default);
+              } else {
+                return transformerFn(exports);
+              }
+              return exports;
+            });
+          } catch (e) {
+            logger.error(`Cannot instrument ${moduleName} due to an error in instrumentation: ${e}`);
+            if (e.message) {
+              logger.error(e.message);
+            }
+            if (e.stack) {
+              logger.error(e.stack);
+            }
+          }
+        } else {
+          logger.error(
```

Comment:
Is this a misusage error?

Reply:
the intention is to log an error when the type of the transformer function is not a function, similiar check there in requirehook

Reply:
So it is a misusage log? We just need to ensure that a customer does not run into this.